internals/overlord/servstate/handlers.go

```diff
@@ -115,13 +118,21 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 	}
 
 	currentPlan := m.getPlan()
-	config, ok := currentPlan.Services[request.Name]
-	if !ok {
+	config, configFound := currentPlan.Services[request.Name]
+	if !configFound {
 		return fmt.Errorf("cannot find service %q in plan", request.Name)
 	}
 
+	var workload *workloads.Workload
+	if config.Workload != "" {
+		ws := currentPlan.Sections[workloads.WorkloadsField].(*workloads.WorkloadsSection)
```

Comment:
This will panic if the type isn't right. I know that such case would be an unrecoverable internal error anyway, so panicking isn't wrong necessarily. What I'm suggesting is to make it explicit i.e. own the panic instead of leaving it up to interpretation whether this is meant to panic in the bad case.

Reply:
I believe I suggested we don't need special handling here (to simplify the code). I can see what Paul's saying, but I think that would be a slippery slope, leading to extraneous `if`s and explicit `panic`s in a ton of places if we followed that everywhere. How about a compromise: leave the code as is but add a comment like `// Will panic if we used an incorrect type (programmer error)`?